### DISCUSSION TITLE
Resizing the image without changing dimensions

### DISCUSSION THREAD

**Discussion by Sparkenstein** [image-rs/image#1794] [Q&A]
Hey, 
I thought discussions would be a better place to ask this, the `resize` function will change the dimensions of image, but what I want is just compressing the image without changing dimensions, WHILE converting the image formats as well.
 
I am working on something like https://squoosh.app/  
Any help is appreciated 🙏

**Comment by asonix** [comment]
I think "resize" is not the thing you're looking for. You want to reduce the _file size_, not the _image size_. In that case, you may want to look at specific encoders like jpeg's [`new_with_quality`](https://docs.rs/image/latest/image/codecs/jpeg/struct.JpegEncoder.html#method.new_with_quality) or png's [`new_with_quality`](https://docs.rs/image/latest/image/codecs/png/struct.PngEncoder.html#method.new_with_quality)

**Comment by szabgab** [comment]
I am not sure if this is what you are looking for, but I wrote an example [here](https://rust.code-maven.com/resize-image)
